src/Microsoft.Azure.WebJobs.Extensions.SignalRService.Protocols/SignalRExtensionProtocols.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Azure.EventHubs;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService.Protocols
+{
+    public class SignalRExtensionProtocols : ISignalRExtensionProtocols
+    {
+        public int Version => 1;
+
+        public bool TryParseMessage(EventData input, out SignalRExtensionMessage message)
+        {
+            if (!input.Properties.TryGetValue(SignalRExtensionProtocolConstants.MessageType, out var messageType))
+            {
+                message = null;
+                return false;
+            }
+
+            switch (messageType)
+            {
+                case SignalRExtensionProtocolConstants.CloseConnectionType:
+                    message = new CloseConnectionExtensionMessage(input);
+                    return true;
+                case SignalRExtensionProtocolConstants.OpenConnectionType:
+                    message = new OpenConnectionExtensionMessage(input);
+                    return true;
+                case SignalRExtensionProtocolConstants.InvocationType:
+                    message = new InvocationExtensionMessage(input);
+                    return true;
+                default:
+                    message = null;
+                    return false;
+            }
+        }
+
+        public EventData BuildMessage(int messageType, string hub, string connectionId, object body)
+        {
+            JObject o = JObject.FromObject(body);
+            
+            var date = new EventData(Encoding.UTF8.GetBytes(o.ToString()));
```

Comment:
date -> data

Reply:
Should version be serialized as well?

Reply:
It's good to add version into properties as well, but do we need to prevent parsing messages if the version isn't equal.